src/tests/fuzz/fuzz_applayerparserparse.c

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file
+ * @author Philippe Antoine <contact@catenacyber.fr>
+ * fuzz target for AppLayerProtoDetectGetProto
+ */
+
+
+#include "suricata-common.h"
+#include "app-layer-detect-proto.h"
+#include "flow-util.h"
+#include "app-layer-parser.h"
+#include "util-unittest-helper.h"
+
+#define HEADER_LEN 6
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+AppLayerParserThreadCtx *alp_tctx = NULL;
+
+/* input buffer is structured this way :
+ * 6 bytes header,
+ * then sequence of buffers separated by magic bytes 01 D5 CA 7A */
+
+/* The 6 bytes header is
+ * alproto
```

Comment:
I could see value in a (set of) target(s) that allow us to specifically test a app-layer parser. For example when we introduce a new parser, or a large change (e.g. big libhtp update for http) we really want to hammer it for some time.

Reply:
This is interesting. Here are a few elements for thinking.

This target is quite easy to get modified to test a specific app-layer.
We can copy/paste the code (n times) and enforce the value.
Or we can use an environment variable, or maybe an argument with https://llvm.org/docs/LibFuzzer.html#startup-initialization but I am not sure if we can fit this into oss-fuzz
Would that fit your use case ?

This fuzz target allows to fuzz a new parser without changes
Otherwise, you need to add a corresponding new fuzz target when you add a new parser...

I think that the fuzzing engines are smart enough, when there is one big change, to "hammer" it (because they find new code coverage)

Reply:
Is there a way to find out in what alproto a running fuzzer spends most of its time? As an example, I would like to 'hammer' a new protocol for at least a week before doing a release. However if it turns out the fuzzer spends almost all its time in http/libhtp, the new protocol may have hardly been fuzzed.

I do think using an env var could be good enough. It would allow the manual 'hammering' while otherwise relying on the smartness of the coverage based approach.

Reply:
> Is there a way to find out in what alproto a running fuzzer spends most of its time?

oss-fuzz generates a nice coverage report
You can do it locally with your corpus as well.

> using an env var could be good enough. It would allow the manual 'hammering'

Yes that would be manual.
I will propose something like that

Reply:
> > Is there a way to find out in what alproto a running fuzzer spends most of its time?
> 
> oss-fuzz generates a nice coverage report
> You can do it locally with your corpus as well.

Can you update the bundled doc with instructions on how to generate this?

Reply:
Ok, doc is here : https://google.github.io/oss-fuzz/advanced-topics/code-coverage/